### PR TITLE
NIAD-182: openapi docs

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

**Review by Broom94 — DISMISSED** [review]

File: outbound/request/tests/test_amendment_payloads.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+
+def _create_address_minimal() -> Address:
+    a = Address()
+    a.line = ['current house name', 'current number or road name', 'current locality', 'current town', 'current county']
+    a.postalCode = 'SO15 2GB'
+    return a
+
+
+class TestAcceptancePayloads(unittest.TestCase):
```
Comment (michael-kainos) [inline outbound/request/tests/test_amendment_payloads.py]: is this class name meant to be amendment

File: outbound/request/tests/test_deduction_payloads.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+def _create_org_ref(identifier_value: str):
+    ref = FHIRReference()
+    ref.type = Organization.resource_type
+    identifier = Identifier()
+    identifier.value = identifier_value
+    ref.identifier = identifier
+    return ref
+
+
+class TestAcceptancePayloads(unittest.TestCase):
```
Comment (michael-kainos) [inline outbound/request/tests/test_deduction_payloads.py]: class name deduction here?

File: outbound/request/tests/test_removal_payloads.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+def _create_org_ref(identifier_value: str):
+    ref = FHIRReference()
+    ref.type = Organization.resource_type
+    identifier = Identifier()
+    identifier.value = identifier_value
+    ref.identifier = identifier
+    return ref
+
+
+class TestAcceptancePayloads(unittest.TestCase):
```
Comment (michael-kainos) [inline outbound/request/tests/test_removal_payloads.py]: removal class name here?

File: specification/components/examples/requests/NHAIS/acceptance-birth-max.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        "current house name",
+        "current number or road name",
+        "current locality",
+        "current town",
+        "current county"
+      ],
+      "postalCode": "SO15 2GB"
+    }
+  ],
+  "birthDate": "1990-01-01T09:33:00",
```
Comment (michael-kainos) [inline specification/components/examples/requests/NHAIS/acceptance-birth-max.json]: should max birth date be 1990?

File: specification/components/examples/requests/NHAIS/acceptance-birth-min.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        "current house name",
+        "current number or road name",
+        "current locality",
+        "current town",
+        "current county"
+      ],
+      "postalCode": "SO15 2GB"
+    }
+  ],
+  "birthDate": "1990-01-01T09:33:00",
```
Comment (michael-kainos) [inline specification/components/examples/requests/NHAIS/acceptance-birth-min.json]: does the time for min need be be at the boundary:
birthDate": "1990-01-01T00:01:00

File: specification/components/examples/requests/NHAIS/removal.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (michael-kainos) [inline specification/components/examples/requests/NHAIS/removal.json:1]: is no operation definition needed? e.g: delete?

File: specification/components/schemas/HumanName.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      type: string
+      example: Jane
+  family:
+    type: string
+    description: Family name (often called Surname).
+    example: Smith
+  prefix:
+    type: array
+    description: Name prefixes, titles, and prenominals.
+    example: [Mrs]
```
Comment (michael-kainos) [inline specification/components/schemas/HumanName.yaml:43]: do we need enums for this?

File: specification/components/schemas/codeable/DeathNotificationStatus.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
+type: object
+description: Death Notification Status.
+properties:
+  coding:
+    type: array
+    minItems: 1
+    maxItems: 1
```
Comment (michael-kainos) [inline specification/components/schemas/codeable/DeathNotificationStatus.yaml:7]: is this counting from 0? if not should this be 2?

File: specification/components/schemas/codeable/Language.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
+type: object
+description: Human language.
+required:
+  - coding
+properties:
+  coding:
+    type: array
+    description: Exactly one language code.
+    minItems: 1
+    maxItems: 1
```
Comment (michael-kainos) [inline specification/components/schemas/codeable/Language.yaml:10]: opposite to previous comment, is this counting from 0? if so should this be 0?

File: specification/components/schemas/codeable/NHSNumberVerificationStatus.yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
+type: object
+description: NHS Number Verification Status Indicator.
+required:
+  - coding
+properties:
+  coding:
+    type: array
+    minItems: 1
+    maxItems: 1
```
Comment (michael-kainos) [inline specification/components/schemas/codeable/NHSNumberVerificationStatus.yaml:9]: same here ?

File: specification/components/schemas/codeable/RelatedPersonRelationshipType.yaml [hunk 11]
```diff
@@ -0,0 +1,23 @@
+type: object
+properties:
+  coding:
+    type: array
```
Comment (michael-kainos) [inline specification/components/schemas/codeable/RelatedPersonRelationshipType.yaml:4]: no required objects for this array?

File: specification/components/examples/requests/NHAIS/acceptance-birth-max.json (hunk 4 above)
Comment (mattd-kainos) [reply specification/components/examples/requests/NHAIS/acceptance-birth-max.json]: max as an maximum number of fields (including all optional) not maximum values

File: specification/components/examples/requests/NHAIS/acceptance-birth-min.json (hunk 5 above)
Comment (mattd-kainos) [reply specification/components/examples/requests/NHAIS/acceptance-birth-min.json]: as above, max/min aren't ranges but required/optional. Maybe I should rename the files?

File: specification/components/schemas/extensions/OperationIdentifier.yaml [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  extension:
+    type: object
+    description: "Asynchronous operation identifiers. TODO: Could optionally include NHAIS-specific IDs if needed to support queries from HA"
+    required:
+      - operationId
+    properties:
+      operationId:
+        type: string
+        description: The identifier (UUID) of the accepted transaction
+        pattern: "[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}"
```
Comment (michael-kainos) [inline specification/components/schemas/extensions/OperationIdentifier.yaml:21]: is the generated UUID always containing a-f and A-F alphabetical characters? are these not randomised using all alphabetical characters?

File: specification/components/examples/requests/NHAIS/removal.json (hunk 6 above)
Comment (mattd-kainos) [reply specification/components/examples/requests/NHAIS/removal.json:1]: My understanding is that the URI defines the operation type e.g. /fhir/Patient/12345/$nhais.removal would perform an NHAIS removal operation on Patient id 12345 with parameters supplied in the payload

**Review by michael-kainos — COMMENTED** [review]
@mattd-kainos please see review comments

File: specification/components/schemas/HumanName.yaml (hunk 7 above)
Comment (mattd-kainos) [reply specification/components/schemas/HumanName.yaml:43]: This is from the PDS API

File: specification/components/examples/requests/NHAIS/acceptance-birth-min.json (hunk 5 above)
Comment (michael-kainos) [reply specification/components/examples/requests/NHAIS/acceptance-birth-min.json]: > as above, max/min aren't ranges but required/optional. Maybe I should rename the files?

Understood, yeah, rename may make it clearer, thanks

File: specification/components/schemas/codeable/Language.yaml (hunk 9 above)
Comment (mattd-kainos) [reply specification/components/schemas/codeable/Language.yaml:10]: PDS API

File: specification/components/schemas/codeable/NHSNumberVerificationStatus.yaml (hunk 10 above)
Comment (mattd-kainos) [reply specification/components/schemas/codeable/NHSNumberVerificationStatus.yaml:9]: PDS API

File: specification/components/schemas/codeable/RelatedPersonRelationshipType.yaml (hunk 11 above)
Comment (mattd-kainos) [reply specification/components/schemas/codeable/RelatedPersonRelationshipType.yaml:4]: PDS API

File: specification/components/schemas/extensions/OperationIdentifier.yaml (hunk 12 above)
Comment (mattd-kainos) [reply specification/components/schemas/extensions/OperationIdentifier.yaml:21]: hex, a-f only. This is the correct pattern for a UUID

File: specification/components/schemas/codeable/DeathNotificationStatus.yaml (hunk 8 above)
Comment (mattd-kainos) [reply specification/components/schemas/codeable/DeathNotificationStatus.yaml:7]: PDS API

**Review by michael-kainos — APPROVED** [review]

**Review by Broom94 — APPROVED** [review]

**Pull request merged**